pkg/api/steve/supportconfigs/generator.go

```diff
@@ -24,34 +25,114 @@ import (
 
 const (
 	// Endpoint The endpoint that this URL is accessible at - used for routing and for SARs
-	Endpoint            = "/v1/generateSUSERancherSupportConfig"
+	Endpoint = "/v1/generateSUSERancherSupportConfig"
+	// NOTE: the name of the configmap will be the same for both Bring-Your-Own (BOY) and
+	// Pay-As-You-Go (PAYG) offerings. However, they will be in different namespaces to avoid
+	// clashing.
 	cspAdapterConfigmap = "csp-config"
 	tarContentType      = "application/x-tar"
 	logPrefix           = "support-config-generator"
 )
 
 var errNotFound = errors.New("not implemented")
 
-// Handler implements http.Handler - and serves supportconfigs (tar file which contains support relevant information)
-type Handler struct {
+type cspAdapterInterface interface {
+	GetRelease(chartNamespace string, chartName string) (*release.Release, error)
+}
+
+type cspConfigInterface interface {
+	authorize(cspNamespace string, r *http.Request) (bool, error)
+	getCSPConfig(cspNamespace string) (map[string]interface{}, error)
+}
+
+type CSPConfig struct {
 	ConfigMaps           v1.ConfigMapInterface
 	SubjectAccessReviews authv1.SubjectAccessReviewInterface
-	adapterUtil          *cspadapter.ChartUtil
+}
+
+// authorize checks to see if the user can get the csp adapter configmap. Returns a bool (if the user is authorized)
+// and optionally an error
+func (c *CSPConfig) authorize(cspNamespace string, r *http.Request) (bool, error) {
```

Comment:
I don't think that this makes sense to move to a separate struct. I am assuming that you did this for testing purposes, but this is unnecessary - both `ConfigMaps` and `SubjectAccessReviews` are interfaces and you can mock those directly. The additional benefit there is that you can include the code in this authorize function as part of your testing - in the current form, none of the code in authorize is being tested.

As far as using mocks of those interfaces, I would recommend the following path:
- The configmap interface has a mock here: https://github.com/rancher/rancher/blob/3319bfb6c0947ef262c716ccc970c30509392d57/pkg/generated/norman/core/v1/fakes/zz_generated_config_map_mock.go#L735.  Basically, construct one of these, and then replace the `GetFunc` with the test behavior that you want to mock out (i.e. return a configmap/error).
- The SubjectAccessReviews is a native k8s type, and has a fake as part of client-go. You can see an example of something similar in this test here: https://github.com/rancher/rancher/blob/3319bfb6c0947ef262c716ccc970c30509392d57/pkg/serviceaccounttoken/secret_test.go#L106. Basically the process is as follows:
  - Create a sar that has the response you want (i.e. allowed or denied)
  - Create a `fake.NewClientset` that optionally takes this SAR as input (any input you give new should be returned, but you may need to make a reactor like you see in the linked test, particularly if you want to handle error cases)
  - Extract the SAR interface and use it for the handler using `client.AuthorizationV1().SubjectAccessReviews()`
- Alternatively, the SAR interface only has one method (create) so you could bypass the k8s library and make a testing struct that only implements create. But I would recommend the library since it's here and we have examples of it.

Reply:
Will do.